### PR TITLE
style: change design pattern structure

### LINKED ISSUE #45: Bad tree structure for Design pattern sheets
Currently, the sheets on the different design patterns do not respect the repo structure.

Firstly, the [readme in `prog_object-orientated/design_patterns`](https://github.com/readthedocs-fr/notions/blob/master/prog_orientee_objet/design_patterns/README.md) has not been translated into English. It should contain both languages, since at least one of the sheets has been translated.

Moreover, the path construction differs between the two languages. 
In the French version, each notion has its own folder, even if it is not necessary.
```
prog_object-orientated/design_patterns/fr/observateur/OBSERVATEUR_PATTERN.md
prog_object-orientated/design_patterns/fr/visiteur/VISITEUR_PATTERN.md
```
However, the English version has no sub-folder, the sheet is directly accessible after the choice of language.
```
prog_object-orientated/design_patterns/en/OBSERVER_PATTERN.md
```

**What should we do then?**

If we have to respect the [current repo structure](https://github.com/readthedocs-fr/notions/blob/master/CONTRIBUTING.md#organisation-1), French notions must be moved to the `fr/` folder.
Indeed for the moment, the language is always the last one in the path of the different courses, as you can see (except for exercices):
```
exercices/fr/java/fonction_composee/CORRECTION.md
exercices/fr/java/fonction_composee/ENONCE.md
java/exceptions/fr/EXCEPTIONS_JAVA.md
java/généricité/listes_wildcard/fr/LISTES_WILDCARD.md
java/interfaces_fonctionnelles/en/FUNCTIONAL_INTERFACES.md
java/interfaces_fonctionnelles/fr/INTERFACES_FONCTIONNELLES.md
java/tests_unitaires/fr/TESTS_UNITAIRES_JAVA.md
prog_fonctionnelle/curryfication/fr/CURRYFICATION.md
prog_orientee_objet/abstraction/en/ABSTRACTION.md
prog_orientee_objet/abstraction/fr/ABSTRACTION.md
python/starter_kit/fr/START_WELL.md
```
Also it will be necessary to translate the readme.

### PR DESCRIPTION
**Pull request by LukaMrt**
fix #45 

According to the recommendations of @fusetim

### CODE REVIEW

File: prog_orientee_objet/design_patterns/fr/VISITEUR_PATTERN.md [hunk 1]
```diff
@@ -10,6 +10,8 @@
 
 ## Une mise en situation pour mieux comprendre
 
+> L'entièreté du code est disponible dans le dossier [code_samples/visiteur](../code_samples/visitor).
```
Comment (fusetim) [inline prog_orientee_objet/design_patterns/fr/VISITEUR_PATTERN.md:13]:
```suggestion
> L'entièreté du code est disponible dans le dossier [code_samples/visitor](../code_samples/visitor).
```

**Comment by fusetim** [comment]
Seems good to me, otherwise.

File: prog_orientee_objet/design_patterns/README.md [hunk 2]
```diff
@@ -6,10 +6,23 @@
...
+
+# Design Patterns
+
+> French course authored by [Antoine Tran](https://github.com/Tran-Antoine), [Alexandre Orékhoff](https://github.com/Hokkaydo)
+
+> README translation authored by [Luka Maret](https://github.com/LukaMrt)
+
+A design pattern is a way to build a part of a software which is recognized as a good response to a recurrent software design problem.
+
+Each chapter presenting a pattern will be represented by a markdown file with the course content and a folder containing the final code which is used in support in the course. The code will be available in one or more languages. All folders containing code are placed in [code_examples](code_samples) folder. The course usually presents a pattern with examples in one language.
```
Comment (Tran-Antoine) [inline prog_orientee_objet/design_patterns/README.md:24]:
```suggestion
Each chapter presenting a pattern will be represented by a markdown file with the course content and a folder containing the final code which is used in support of the course. The code will be available in one or more languages. All folders containing code will be placed in the [code_examples](code_samples) folder. The course usually presents a pattern with examples in one language.
```

File: prog_orientee_objet/design_patterns/README.md [hunk 3]
```diff
@@ -6,10 +6,23 @@
...
+
+## Table of contents
+
+- [Observer pattern - definition, problem and response](en/OBSERVER_PATTERN.md)
```
Comment (Tran-Antoine) [inline prog_orientee_objet/design_patterns/README.md:28]:
```suggestion
- [Observer pattern - definition, issue and resolution](en/OBSERVER_PATTERN.md)
```

Comment (Tran-Antoine) [reply prog_orientee_objet/design_patterns/README.md:28]: Suggested change for slightly more common and precise terms

**Review by Tran-Antoine — CHANGES_REQUESTED** [review]

**Review by Tran-Antoine — APPROVED** [review]

File: prog_orientee_objet/design_patterns/README.md [hunk 4]
```diff
@@ -6,10 +6,23 @@
...
+
+# Design Patterns
+
+> French course authored by [Antoine Tran](https://github.com/Tran-Antoine), [Alexandre Orékhoff](https://github.com/Hokkaydo)
+
+> README translation authored by [Luka Maret](https://github.com/LukaMrt)
+
+A design pattern is a way to build a part of a software which is recognized as a good response to a recurrent software design problem.
+
+Each chapter presenting a pattern will be represented by a markdown file with the course content and a folder containing the final code which is used in support of the course. The code will be available in one or more languages. All folders containing code will be placed in the [code_examples](code_samples) folder. The course usually presents a pattern with examples in one language.
```
Comment (fusetim) [inline prog_orientee_objet/design_patterns/README.md:24]: Please, use the same name folder as the real path.

File: prog_orientee_objet/design_patterns/README.md [hunk 5]
```diff
@@ -6,10 +6,23 @@
 Un design pattern (`patron de conception` en français, bien que le terme soit peu courant) est une manière de programmer une partie d'un logiciel, reconnu comme bonne pratique en réponse à un problème récurrent de conception d'un logiciel.
 Ce cours présentera différents patterns, expliquant leur utilisation ainsi que le problème qu'ils cherchent à résoudre.
 
-Chaque chapitre présentant un pattern possèdera un fichier markdown contenant le contenu du cours, ainsi qu'un dossier contenant le code final (disponible dans un ou plusieurs langages)
-utilisé comme support. Le cours présente généralement le pattern en s'appuyant sur un langage en particulier.
+Chaque chapitre présentant un pattern possèdera un fichier markdown contenant le contenu du cours, ainsi qu'un dossier contenant le code final (disponible dans un ou plusieurs langages) utilisé comme support. Tous les dossiers contenant le code des exemples sont placés dans le dossier [code_examples](code_samples). Le cours présente généralement le pattern en s'appuyant sur un langage en particulier.
```
Comment (fusetim) [inline prog_orientee_objet/design_patterns/README.md:9]: Same as https://github.com/readthedocs-fr/notions/pull/46#discussion_r479683031

**Review by fusetim — APPROVED** [review]
Seems good to merge.

File: prog_orientee_objet/design_patterns/fr/OBSERVATEUR_PATTERN.md [hunk 6]
```diff
@@ -207,9 +207,9 @@ Impeccable !
...
+Par ailleurs, nous retrouvons dans ce code un bel exemple du principe `Ouvert/Fermé` *(Open/Closed en anglais)* symbolisé par la lettre `O` dans la liste de principe dits [SOLID](https://github.com/readthedocs-fr/notions/tree/master/prog_orientee_objet/principes_solid/). En effet, notre classe `Alarm` est ouverte à l'extension mais fermée à la modification. Nous pouvons l'étendre à l'infini en lui ajoutant de plus en plus de composants sans pour autant devoir la modifier. Le système de l'alarme est terminé, nous n'avons plus besoin de le changer. Ainsi, nous sommes sûr de ne jamais mettre en péril le fonctionnement de cette classe lors de l'ajout d'un nouveau dispositif auto-réactif.
```
Comment (Aakodal) [inline prog_orientee_objet/design_patterns/fr/OBSERVATEUR_PATTERN.md:210]:
```suggestion
Par ailleurs, nous retrouvons dans ce code un bel exemple du principe `Ouvert/Fermé` *(Open/Closed en anglais)* symbolisé par la lettre `O` dans la liste de principes dits [SOLID](https://github.com/readthedocs-fr/notions/tree/master/prog_orientee_objet/principes_solid/). En effet, notre classe `Alarm` est ouverte à l'extension mais fermée à la modification. Nous pouvons l'étendre à l'infini en lui ajoutant de plus en plus de composants sans pour autant devoir la modifier. Le système de l'alarme est terminé, nous n'avons plus besoin de le changer. Ainsi, nous sommes sûr de ne jamais mettre en péril le fonctionnement de cette classe lors de l'ajout d'un nouveau dispositif auto-réactif.
```
small typo to fix

**Pull request merged**